lib/ansible/plugins/httpapi/splunk.py

```diff
@@ -0,0 +1,62 @@
+# (c) 2019 Red Hat Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+author: Ansible Security Automation Team
+httpapi : splunk
+short_description: HttpApi Plugin for Splunk
+description:
+  - This HttpApi plugin provides methods to connect to Splunk over a
+    HTTP(S)-based api.
+version_added: "2.8"
+"""
+
+import json
+
+from ansible.module_utils.basic import to_text
+from ansible.errors import AnsibleConnectionFailure
+from ansible.module_utils.six.moves.urllib.error import HTTPError
+from ansible.plugins.httpapi import HttpApiBase
+from ansible.module_utils.connection import ConnectionError
+
+BASE_HEADERS = {
+    'Content-Type': 'application/json',
+}
+
+
+class HttpApi(HttpApiBase):
+    def send_request(self, request_method, path, payload=None):
+        data = json.dumps(payload) if payload else '{}'
+
+        try:
+            self._display_request(request_method)
+            response, response_data = self.connection.send(path, payload, method=request_method, headers=BASE_HEADERS, force_basic_auth=True)
```

Comment:
variable `data` is unused, it should ideally go in place of `payload` in `self.connection.send` call.